### PR TITLE
Feat/node env

### PR DESCRIPTION
**Pull request by profy12**
Should be integrated for a first demo of nodejs integration, will need nodejs teacher feedback before close and make documentation.

### CODE REVIEW

File: containers-config/terminal/Dockerfile [hunk 1]
```diff
@@ -18,6 +20,12 @@ RUN apt update \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/*
 
+RUN curl -fsSL https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - && \
+    echo "deb https://deb.nodesource.com/node_20.x bullseye main" | tee /etc/apt/sources.list.d/nodesource.list
+
+RUN apt-get update && apt-get install -y nodejs && npm install -g n && \
+    rm -rf /var/lib/apt/lists/* && rm -rf /tmp/* && apt-get clean
+
```
Comment (clementgoclock) [inline containers-config/terminal/Dockerfile:28]: We can just install n first from repo https://github.com/mklement0/n-install?tab=readme-ov-file#installation-options and then use n for nodejs installation.
Remove the need for curl/curling somehting and doing apt operation yet after the first one

**Review by clementgoclock — CHANGES_REQUESTED** [review]

File: containers-config/terminal/Dockerfile (hunk 1 above)
Comment (profy12) [reply containers-config/terminal/Dockerfile:28]: the curl is just for the gpg key, n is an option for the future and allow students/teacher to installe the version they want if the version 20 in the image has not been updated.

Comment (clementgoclock) [reply containers-config/terminal/Dockerfile:28]: No need for GPG key and adding node repo when you pass this [script](https://raw.githubusercontent.com/mklement0/n-install/stable/bin/n-install) to bash through wget and get `n` installed. Then you can install node20.

```Dockerfile
RUN wget -O - https://raw.githubusercontent.com/mklement0/n-install/stable/bin/n-install | bash

RUN n lts
```

Comment (profy12) [reply containers-config/terminal/Dockerfile:28]: I prefer to use official repository of an application than a script downloaded without any security check. But maybe we can put the GPG key directly in the dockerfile.
Using n is an option in the future to update Node if needed the time the official image is updated with the good version, but it is I think a local use in the lifecyle of the container and not in the lifecycle of the image.

**Review by clementgoclock — APPROVED** [review]
gogo

**Pull request merged**